### ISSUE TITLE
Suggested features for working with regular partial + occasional full backups

### ISSUE THREAD

**Issue by crowbarz** [sabeechen/hassio-google-drive-backup#160]
I wanted to suggest several features that I think would make the addon's abilities to create regular partial snapshots with additional manually created full snapshots more useful:

- a way to configure the addon to _only_ back up snapshots that it created itself, or
- be able to manually mark specific snapshots so that they _aren't_ uploaded to Drive (though this would need some way to mark this in a timely manner - the addon seems to back up new snapshots ~immediately), or
- set a snapshot size limit, above which the addon doesn't automatically back them up to Drive. This could also throw a big red error that if accepted mark the snapshots as ignored for upload.

In my setup, the addon is configured to create daily ~160MB partial snapshots that exclude large, mostly stateless and/or problematic addons like vscode and mariadb, then uploading the most recent 7 days + 4 weeks of backups to Drive via generational backup. This all works quite well.

Just before upgrading addons and/or HA itself, I always create a full backup (after stopping mariadb to avoid home-assistant/hassio-addons#667, I know it's fixed but call me paranoid :) which are closer to 2GB in size. Given Drive has limited capacity, I don't want these full backups to be uploaded there.

What I currently do now to avoid filling my Drive with full snapshots is:
1. stop the automatic upload to Drive in the addon settings
2. create a full snapshot via hassio
3. perform the upgrade(s) as required
4. once I decide that I probably don't need to roll back, usually about a week later, download/delete/move out of sight the full snapshot
5. remember to go back into the addon settings to re-enable automatic Drive upload.

Any of the suggested features at the top of this bug would make it easier to leave the full snapshots as is locally alongside the partial snapshots, without them being uploaded to Drive by the addon. The generational backup deletion would potentially also look after deleting the full snapshots some time later when they are truly not going to be useful.

Thanks ... and thanks also for a fantastically useful addon that has vastly improved the recoverability of my HA setup!

**Comment by sabeechen** [comment]
Thanks for the suggestions, I'd like to implement at least some part of this.  Also especially thank you for explaining the workflow you'd like to use it in, that helps me understand how best to implement things.
What im thinking of is to add to the "Snapshot Now" menu in the addon UI the ability to:
- Start a full or partial snapshot (in addition to the way it works now, which is to create a snapshot with your "default" settings)
- Add a checkbox for "Disable upload to Google Drive" which would cause the snapshot to never get uploaded to Google Drive after its' created.

Then I think you could accomplish what you're trying to do without having to stop/restart the addon  so long as you created all of your snapshots inside of the addon.  I'd also have to add an option somewhere to "Resume Google Drive upload" for snapshots created in this way.  I think this would be the best compromise between being:
- Not too difficult to implement
- Works for your use case
- Probably also usefull for others

**Comment by dethpickle** [comment]
If you're still considering this, can I suggest also offering a slightly different approach?

I might have a couple days every several months where I'll jump in and start making config changes to my Hassio.  But more often I'm really just absorbing the core and add-in updates as they come.  My preferred backup strategy would be to retain a week of daily backups locally and only upload once a week.  I'd also like to retain the last X _versions_ backup {version_ha} uploaded.

Basically, I'm less concerned about backups by date and am more concerned by version.  When something goes wrong, it's usually because I've upgraded and broken something.  Worst case, I don't notice it immediately.

I actually was thinking this type of option might have been in generational backups, but they seem date based exclusively.

**Comment by sabeechen** [comment]
Its been quite a while since I comented on this thread, but I just released v0.104.1 of the addon which includes a feature to ignore any snapshots the addon didn't create.  This was added in repsonse to HA's recent change to create snapshots with every addon/HA upgrade, but will also allow you to accomplish what you originally posted about.

If you run into any trouble with the feature, please let me know.
